utils/eth1.go

```diff
@@ -346,6 +347,110 @@ func formatAmount(amount *big.Int, unit string, digits int, maxPreCommaDigitsBef
 	return template.HTML(fmt.Sprintf("<span%s>%s%s</span>", tooltip, preComma, displayUnit))
 }
 
+// NewFormat returns a new HTML-formatted string representing the given Wei big.Int amount, with the specified number of digits after the decimal point and number of digits hidden on mobile views.
+//
+// Parameters:
+//   - `amount` : the Wei(!) amount to be formatted.
+//   - `unit` : the unit in which the amount should be formatted. Supported values are ETH and GWei.
+//   - `digits` : the number of digits after the decimal point to be displayed.
+//   - `mobileDigitsHide` : the number of trailing decimal digits to hide on mobile views.
+//   - `maxPreCommaDigitsBeforeTrim` : the maximum number of digits before the decimal point that should be displayed before trimming excess decimal digits. If this is set to 0, no digits are trimmed.
+//
+// **NOTE** Every number is rounded down
+//
+// TODO Replace every formatAmount with this
+func NewFormat(amount *big.Int, unit string, digits int, mobileDigitsHide int, maxPreCommaDigitsBeforeTrim int) template.HTML {
```

Comment:
I may have found some unintentional behavior in this function. 

Here is my testcode:
```
myTest := utils.NewFormat(big.NewInt(1234567890123456789), "ETH", 7, 2, 0)
logger.Info(myTest)
myTest = utils.NewFormat(big.NewInt(1234000000000000000), "ETH", 7, 2, 0)
logger.Info(myTest)
```
This will show the following logs:
> msg="<span data-toggle=\"tooltip\" data-placement=\"top\" title=\"1.234567890123456789\">1.23456<span class=\"d-none d-sm-inline\">78</span> ETH</span>"
> msg="<span data-toggle=\"tooltip\" data-placement=\"top\" title=\"1.234\">1.2<span class=\"d-none d-sm-inline\">34</span> ETH</span>"

**mobileDigitsHide** is used to limit the digital points on mobile. But it works differently based on the passed **amount**. If the **amount** results in trailing 0 that are cut off, **mobileDigitsHide** hides different decimal point positions.

In the first test case, the decimal points on positions 6 and 7 have been hidden. This is as I would expect it as I want to show 7 decimal places (see **digits**) but hide 2 on mobile. The result of "1.23456(78)" (78 hidden on mobile) seems perfectly fine.
In the second test case, the decimal points on positions 2 and 3 have been hidden resulting in only 1 decimal place on mobile: "1.2(34)"

A possible solution would be to add trailing zeros but I am not sure how all this should behave.

This whole comment might be super confusing so I suggest we talk about this face2face.

Reply:
After talking with the designer, having a different number of decimal places on desktop and mobile views was not intended (all ETH amounts should display 5 decimal places). Everything related to hiding digits on mobile was removed.